src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -2150,12 +2174,15 @@ def _get_network_plugin(self, enable_validation: bool = False) -> Union[str, Non
             ) = self._get_pod_cidr_and_service_cidr_and_dns_service_ip_and_docker_bridge_address_and_network_policy(
                 enable_validation=False
             )
+            network_plugin_mode = self._get_network_plugin_mode(enable_validation=False)
             if network_plugin:
-                if network_plugin == "azure" and pod_cidr:
+                if network_plugin == "azure" and pod_cidr and network_plugin_mode != "overlay":
                     logger.warning(
-                        'When using `--network-plugin azure` without the preview feature '
-                        '`--network-plugin-mode overlay` the provided pod CIDR "%s" will be '
-                        'overwritten with the subnet CIDR.', pod_cidr
+                        'The provided pod CIDR "%s" will be overwritten with the node subnet CIDR. '
+                        'To use a pod CIDR, please specify network plugin mode `overlay` or '
+                        'use network plugin `kubenet`. For more information about Azure CNI '
+                        'Overlay please see https://aka.ms/aks/azure-cni-overlay. This warning '
+                        'will become an error in a future CLI release.', pod_cidr
```

Comment:
```suggestion
                        'will become an error in the Build sprint release.', pod_cidr
```

Reply:
do customers know what "the Build sprint" release is? I'm not even sure if I know what what that means?

Reply:
so "Build" as in the conference

> Generally speaking, azure-cli only allows breaking change in the corresponding release of Build and Ignite

from @FumingZhang